src/libpmempool/replica.c

```diff
@@ -528,28 +598,153 @@ check_checksums_and_signatures(struct pool_set *set,
 			if (!util_checksum(hdr, sizeof(*hdr), &hdr->checksum, 0,
 					POOL_HDR_CSUM_END_OFF)) {
 				ERR("invalid checksum of pool header");
-				rep_hs->part[p] |= IS_BROKEN;
+				rep_hs->part[p].flags |= IS_BROKEN;
 			} else if (util_is_zeroed(hdr, sizeof(*hdr))) {
-					rep_hs->part[p] |= IS_BROKEN;
+					rep_hs->part[p].flags |= IS_BROKEN;
 			}
 
 			enum pool_type type = pool_hdr_get_type(hdr);
 			if (type == POOL_TYPE_UNKNOWN) {
 				ERR("invalid signature");
-				rep_hs->part[p] |= IS_BROKEN;
+				rep_hs->part[p].flags |= IS_BROKEN;
 			}
 		}
 	}
 	return 0;
 }
 
 /*
- * check_badblocks -- (internal) check if replica contains bad blocks
+ * redo_log_exist -- set name of the redo log file and check if it exists
  */
 static int
-check_badblocks(struct pool_set *set, struct poolset_health_status *set_hs)
+redo_log_exist(const char *file, struct part_health_status *part_hs)
 {
-	LOG(3, "set %p, set_hs %p", set, set_hs);
+	LOG(3, "file %s part_health_status %p", file, part_hs);
+
+	const char *bbs_suffix = "_badblocks.txt";
+
+	size_t len_path = strlen(file) + strlen(bbs_suffix);
+	char *path = Zalloc(len_path + 1);
+	if (!path) {
+		ERR("!Zalloc");
+		return -1;
+	}
+
+	path = strncpy(path, file, strlen(file));
+	if (!path) {
+		ERR("!strncpy");
+		goto error_free_all;
+	}
+
+	path = strncat(path, bbs_suffix, strlen(bbs_suffix));
+	if (!path) {
+		ERR("!strncat");
+		goto error_free_all;
+	}
+
+	part_hs->redo_log = path;
+	part_hs->exist_redo_log = (os_access(path, F_OK) == 0);
+
+	return 0;
+
+error_free_all:
+	Free(path);
+
+	return -1;
+}
+
+/*
+ * redo_log_save_badblocks -- save bad blocks in the redo log
+ *                            before clearing them
+ */
+static int
+redo_log_save_badblocks(const char *file, struct part_health_status *part_hs)
```

Comment:
What happens if this is interrupted halfway? We should also store some kind of a finish flag ;)
I wouldn't want people to deal with "wrong format of the redo log file" type of errors. It should all be automatic.